ui/src/components/admin/navigation/Sidebar.js

```diff
@@ -0,0 +1,270 @@
+];
+
+const SidebarWithHeader = ({ children, ...rest }) => {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+
+  return (
+    // <Box>
+    <>
+      <SidebarContent
+        {...rest}
+        onClose={() => onClose}
+        display={{ base: "none", md: "block" }}
+      />
+      <Drawer
+        {...rest}
+        autoFocus={true}
+        isOpen={isOpen}
+        placement="left"
+        onClose={onClose}
+        returnFocusOnClose={false}
+        onOverlayClick={onClose}
+        size="full"
+      >
+        <DrawerContent {...rest}>
+          <SidebarContent {...rest} onClose={onClose} />
+        </DrawerContent>
+      </Drawer>
+      {/* mobilenav */}
+      <MobileNav onOpen={onOpen} {...rest} />
+      <Box {...rest} ml={{ base: 0, md: 60 }} p="4">
+        {children}
+      </Box>
+    </>
+    // {/* </Box> */}
+  );
+};
+
+export default SidebarWithHeader;
+
+// interface SidebarProps extends BoxProps {
+//   onClose: () => void;
+// }
+
+const SidebarContent = ({ onClose, ...rest }) => {
+  const { toggleColorMode } = useColorMode();
+
+  return (
+    <Box
+      transition="3s ease"
+      bg={useColorModeValue("white", "gray.900")}
+      borderRight="1px"
+      borderRightColor={useColorModeValue("gray.200", "gray.700")}
+      w={{ base: "full", md: 60 }}
+      pos="fixed"
+      h="full"
+      {...rest}
+    >
+      <Flex h="20" alignItems="center" mx="8" justifyContent="space-between">
+        <Text fontSize="2xl" fontFamily="monospace" fontWeight="bold">
+          HMCC-HK
+        </Text>
+        <CloseButton display={{ base: "flex", md: "none" }} onClick={onClose} />
+      </Flex>
+      {LinkItems.map((link) => (
+        <NavItem
+          key={link.name}
+          icon={link.icon}
+          path={link.path}
+          component={link.component}
+        >
+          {link.name}
+        </NavItem>
+      ))}
+      <Button mx={10} onClick={toggleColorMode}>
+        Toggle Color Mode
+      </Button>
+    </Box>
+  );
+};
+
+// interface NavItemProps extends FlexProps {
+//   icon: IconType;
+//   children: ReactText;
+// }
+const NavItem = ({ icon, children, ...rest }) => {
+  return (
+    <Link
+      href={rest.path}
+      style={{ textDecoration: "none" }}
+      component={rest.component}
+    >
+      <Flex
+        align="center"
+        p="4"
+        mx="4"
+        borderRadius="lg"
+        role="group"
+        cursor="pointer"
+        _hover={{
+          bg: "cyan.400",
+          color: "white",
+        }}
+        {...rest}
+      >
+        {icon && (
+          <Icon
+            mr="4"
+            fontSize="16"
+            _groupHover={{
+              color: "white",
+            }}
+            as={icon}
+          />
+        )}
+        {children}
+      </Flex>
+    </Link>
+  );
+};
+
+// interface MobileProps extends FlexProps {
+//   onOpen: () => void;
+// }
+const MobileNav = ({ onOpen, ...rest }) => {
```

Comment:
is this appropriately called MobileNav? seems like it's not for mobile nav? it's the profile menu to logout, change settings etc. right?